weavetest/tx.go

```diff
@@ -69,3 +71,22 @@ func SequenceID(n uint64) []byte {
 	binary.BigEndian.PutUint64(b, n)
 	return b
 }
+
+// BlockInfo returns a simple BlockInfo struct given just a height and optional time
```

Comment:
:police_car: 

> BlockInfo returns a ~simple~ BlockInfo struct

_simple_ is relative :wink: 


I wonder if instead of two functions `BlockInfo` and `BlockInfoWithChain` wouldn't it be better to provide only the more verbose one.
If this was a Python, only one function would be enough
```python
def block_info(height, ts=None, chain_id="test-chain"):
    pass
```

In Go there are no default arguments. What we can do is similar to C, provide defaults for empty values.
```go
// BlockInfo bla blabla.
// If nil time is provided, current wall clock time is used. If chain ID is not provided, default "test-chain" value is used.
func BlockInfo(height int64, ts *time.Time, chainID string) weave.BlockInfo {
    if ts == nil || ts.Zero() {
        ts = time.Now()
    }
    if chainID ==  "" {
        chainID = "test-chain"
    }
    // ...
}
```
You are allowing to use optional time value (`...time.Time` argument), which is a smart hack, but also a dead end for the API as changing such function notation is extremely hard.

What do you think?

Reply:
This is nice idea. Basically three ways to call and take default args

`BlockInfo(59, &now, "my-chain")` 
`BlockInfo(59, &now, "")` 
`BlockInfo(59, nil, "")`

I was a bit worried about `*time.Time` as we cannot inline calls here:

`BlockInfo(59, &time.Now(), "foo-chain")`

But when looking at the code, we always call `time.Now()` one place and reuse the variable in the calls. I will fix up `weavetest` with this improved syntax.